lib/kb_hmmer/Utils/HmmerUtil.py

```diff
@@ -1789,6 +1923,15 @@ def run_HMMER_Model_Group_Search(self, params):
                         table_genes[genome_ref][cat] = hit_genes_by_genome_and_model[genome_ref][cat]
                         cat_seen[cat] = True
 
+            # adjust to perc all CDS if not raw count
+            if params.get('count_category') and params['count_category'] == 'perc_all':
+                for genome_ref in all_genome_refs:
+                    total_genes = genome_CDS_count_by_ref[genome_ref]
+                    for cat in cats:
+                        if table_data[genome_ref][cat] != 0:
```

Comment:
seems unnecessary. Because if val is 0, after calculation, it will stay 0.

Reply:
but worth to check if `total_genes != 0`

Reply:
added check for genomes without gene calls.  you're right about zero staying zero.  just a habit from not trusting the exact value of a float as much as an int.  I can change it if you feel strongly.